Sources/MapboxCoreNavigation/NavigationEventsManager.swift

```diff
@@ -254,26 +279,27 @@ open class NavigationEventsManager {
     }
     
     func enqueueFeedbackEvent(type: FeedbackType, description: String?) -> UUID? {
-        guard let eventDictionary = (try? navigationFeedbackEvent(type: type, description: description)?.asDictionary()) as [String: Any]?? else { return nil }
+        guard let feedbackEvent = navigationFeedbackEvent(type: type, description: description) else { return nil }
+        let eventDictionary = try? feedbackEvent.asDictionary()
```

Comment:
It’s a bit awkward that this old code had to convert the NavigationEventDetails to a Dictionary just to create a new FeedbackEvent. Maybe there’s a more direct way of doing that, but at least we should add a convenience initializer to FeedbackEvent that takes NavigationEventDetails directly.

Reply:
Thanks, this logic looks strange indeed, hope we will find some time for refactoring after GA.